lib/index.d.ts

```diff
@@ -895,5 +903,60 @@ declare module "react-native-firebase" {
         setUserIdentifier(userId: string): void;
       }
     }
+
+    namespace links {
+      interface Links {
+        /** Creates a standard dynamic link. */
+        createDynamicLink(parameters: LinkConfiguration): Promise<string>;
+        /** Creates a short dynamic link. */
+        createShortDynamicLink(parameters: LinkConfiguration): Promise<string>;
+        /**
+         * Returns the URL that the app has been launched from. If the app was
+         * not launched from a URL the return value will be null.
+         */
+        getInitialLink(): Promise<string | null>;
+        /**
+         * Subscribe to URL open events while the app is still running.
+         * The listener is called from URL open events whilst the app is still
+         * running, use getInitialLink for URLs which cause the app to open
+         * from a previously closed / not running state.
+         * Returns an unsubscribe function, call the returned function to
+         * unsubscribe from all future events.
+         */
+        onLink(listener: (url) => void): () => void;
+      }
+  
+      /**
+       * Configuration when creating a Dynamic Link (standard or short). For
+       * more information about each parameter, see the official Firebase docs:
+       * https://firebase.google.com/docs/reference/dynamic-links/link-shortener
+       */
+      interface LinkConfiguration {
+        link: string,
+        dynamicLinkDomain: string,
+        androidInfo?: {
+          androidLink?: string,
+          androidPackageName: string,
+          androidFallbackLink?: string,
+          androidMinPackageVersionCode?: string,
+        },
+        iosInfo?: {
+          iosBundleId: string,
+          iosAppStoreId?: string,
+          iosFallbackLink?: string,
+          iosCustomScheme?: string,
+          iosIpadBundleId?: string,
+          iosIpadFallbackLink?: string,
+        },
+        socialMetaTagInfo?: {
+          socialTitle: string,
+          socialImageLink: string,
+          socialDescription: string,
+        },
+        suffix?: {
+          option: 'SHORT' | 'UNGUESSABLE',
```

Comment:
Is `UNGUESSABLE` correct here? Shouldn't it just be a string unless I'm missing something...!

Reply:
It is correct. As noted in [the Firebase docs](https://firebase.google.com/docs/reference/dynamic-links/link-shortener), if `suffix` is specified then it must either have value `{ option: "SHORT"
 }` or `{ option: "UNGUESSABLE" }`.

Sample code (works):
```typescript
import firebase from "react-native-firebase";
firebase.app().links().createDynamicLink({
    dynamicLinkDomain: "<app>.goo.gl",
    link: "https://data-to-pass-to-app",
    suffix: {
        option: "UNGUESSABLE",
    },
});
```
Sample code (causes compile error due to disallowed type for `option`):
```typescript
import firebase from "react-native-firebase";
firebase.app().links().createDynamicLink({
    dynamicLinkDomain: "<app>.goo.gl",
    link: "https://data-to-pass-to-app",
    suffix: {
        option: "MADE_UP_STRING",
    },
});
```

Reply:
Confirmed correctly typed as per docs. Will merge as all else seems good.